docs/agent_coordination.md

```diff
@@ -0,0 +1,104 @@
+you can specify: `phases`, `confidence_score_threshold`, and `max_iterations`.
+
+* `phases`: This can be used to specify a list of phase names. The default value is 
+`["Data Collection", "Data Analysis"]`. This list of phase names will be used by the
+`PhasedCoordinatorWiseAgent` to determine the groups of agents that should be executed
+for each phase. Use phase names that make sense for your use case.
+
+* `confidence_score_threshold`: The confidence score threshold (between 0 and 100) to determine
+if the final answer is acceptable. The default value is 85. If the final answer that's been
+obtained after executing all the phases is less than the desired threshold, the
+`PhasedCoordinatorWiseAgent` will attempt to rephrase the original query and try to execute
+the phases again using the improved query to try to get to a better final answer.
+
+* `max_iterations`: This can be used to specify the maximum number of iterations to run
+the phases. The default value is 5. If the final answer that's been obtained after executing 
+all the phases doesn't mean the desired threshold and the maximum number of iterations hasn't
+been reached yet, then the `PhasedCoordinatorWiseAgent` will attempt to rephrase the original
+query and try to execute the phases again.
+
+#### How does a `PhasedCoordinatorWiseAgent` work?
+
+Upon receiving a request, a `PhasedCoordinatorWiseAgent`, will use the names and descriptions
+of the agents from the agents registry to determine the agents that will be needed to solve
+the problem and will group the agents into phases where agents in a phase will be executed
+in parallel. The `PhasedCoordinatorWiseAgent` will then kick off the first phase and once
+all the agents in that phase have finished performing their tasks, the `PhasedCoordinatorWiseAgent`
+will move on to the next phase and so on until all phases have been executed. 
+
+Once all the phases have been executed, the `PhasedCoordinatorWiseAgent` will determine the
+final answer to the original query along with a confidence score.
+
+If the confidence score is equal to or exceeds the `confidence_score_threshold`, the final
+answer will be returned to the client.
+
+If the confidence score is less than the threshold, the `PhasedCoordinatorWiseAgent` will
+attempt to rephrase the original query and try to execute the phases again using the
+improved query to try to get to a better final answer. This process will be repeated until
+the desired confidence score is achieved or until the max number of iterations has been reached.
+
+> **_NOTE:_** The `PhasedCoordinatorWiseAgent` makes use of agent names and descriptions along
+> with phase names to group the agents into phases. It's important to provide good descriptions
+> for your agents to indicate their capabilities and dependencies. For example, if an agent
+> depends on another agent, please indicate this in the agent's description.
```

Comment:
Maybe we want to describe in more detail the algorithm/idea behind agent selection for the phases to be sure users can understand the importance of description. Maybe something like that mentioning they are instruction for the LLM building up the phases could help.

Reply:
Updated the `How does a `PhasedCoordinatorWiseAgent` work?` section.